File: be/src/main/java/sg/toss_sg/models/auth/FlowUserDetails.java
package sg.toss_sg.models.auth;

import java.util.Collection;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class FlowUserDetails implements UserDetails {

    private final int userId;
    private final String name;
    private final Collection<? extends GrantedAuthority> authorities;

    public FlowUserDetails(int userId, String name, Collection<? extends GrantedAuthority> authorities) {
        this.userId = userId;
        this.name = name;
        this.authorities = authorities;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return this.authorities;
    }

    @Override
    public String getPassword() {
        return "SampleFlowPassword";
    }

    @Override
    public String getUsername() {
        return this.name;
    }

    public int getUserId() {
        return userId;
    }

}

File: be/src/main/kotlin/sg/flow/models/auth/FlowUserDetails.kt
package sg.flow.models.auth

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class FlowUserDetails(
        val userId: Int,
        private val name: String,
        private val authoritiesSet: Collection<GrantedAuthority>
) : UserDetails {

    override fun getAuthorities(): Collection<GrantedAuthority> = authoritiesSet

    override fun getPassword(): String = "SampleFlowPassword"

    override fun getUsername(): String = name

    override fun isAccountNonExpired(): Boolean = true
    override fun isAccountNonLocked(): Boolean = true
    override fun isCredentialsNonExpired(): Boolean = true
    override fun isEnabled(): Boolean = true
}
